-1

**Failed Tests**:   Build 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-aa4396/40023/summary.html
**COMMIT**: f285ad2d4a74b495032ba889195c743e6fcf752f
**CMSSW**: CMSSW_14_1_X_2024-06-22-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/45285/40023/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_1_X_2024-06-22-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_1_X_2024-06-22-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/alpaka/1.1.0-0a6641b4bfdf883c8da5b9e8620be504/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/boost/1.80.0-a1544032d9d65904ac2112b6d35bba55/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/lcg/root/6.30.07-5b5d83179d6ed7bc60aa7a5006899eba/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/tbb/v2021.9.0-1dc6d66b4b014f4ae4733b04914ce7d4/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/fmt/8.0.1-258b4791803c34b7e98cf43693e54d87/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/tinyxml2/6.2.0-88fe0ec301baf763fa3c485e5b67ed91/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++17 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/printAnswer.cc.d src/HeterogeneousCore/AlpakaTest/src/alpaka/printAnswer.cc -o tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/printAnswer.cc.o
>> Building alpaka/cuda shared library tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/libHeterogeneousCoreAlpakaTestCudaAsync.so
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++17 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_CUDA_ENABLED -DALPAKA_ACC_GPU_CUDA_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -shared -Wl,-E    -Wl,-z,defs     tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/ES_AlpakaESTestData.cc.o tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/printAnswer.cc.o -o tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/libHeterogeneousCoreAlpakaTestCudaAsync.so -Wl,-E -Wl,--hash-style=gnu -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-06-22-1100/biglib/el8_amd64_gcc12 -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-06-22-1100/lib/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/biglib/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/lib/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/external/el8_amd64_gcc12/lib -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-06-21-2300/lib/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-06-21-2300/biglib/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/cuda/12.4.1-fc5cb0e72dba64b6abbf00089f3a044c/lib64/stubs -L/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_1_X_2024-06-22-1100/static/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/static/el8_amd64_gcc12 -L/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw/CMSSW_14_1_X_2024-06-21-2300/static/el8_amd64_gcc12 -lHeterogeneousCoreAlpakaCore -lHeterogeneousCoreAlpakaCoreCudaAsync -lFWCoreFramework -lFWCoreCommon -lFWCoreServiceRegistry -lDataFormatsCommon -lFWCoreParameterSet -lFWCoreMessageLogger -lDataFormatsProvenance -lFWCoreConcurrency -lFWCorePluginManager -lFWCoreReflection -lHeterogeneousCoreAlpakaInterface -lFWCoreUtilities -lFWCoreVersion -lTree -lNet -lThread -lMathCore -lRIO -lboost_program_options -lCore -lboost_thread -lboost_date_time -lpcre -lbz2 -lcudart -lcudadevrt -lnvToolsExt -luuid -ltbb -llzma -lz -lcuda -lfmt -lcms-md5 -lcrypt -ldl -lrt -lstdc++fs -ltinyxml2
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/cc0hBLMD.ltrans0.ltrans.o: in function `alpaka_cuda_async::alpakatest::printAnswer()':
<artificial>:(.text+0xb2): undefined reference to `alpakatest::HostOnlyType::print()'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestCudaAsync/libHeterogeneousCoreAlpakaTestCudaAsync.so] Error 1
Leaving library rule at src/HeterogeneousCore/AlpakaTest/src/alpaka
Entering library rule at src/HeterogeneousCore/AlpakaTest/src/alpaka
>> Compiling alpaka/rocm src/HeterogeneousCore/AlpakaTest/src/alpaka/ES_AlpakaESTestData.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -D__HIP_PLATFORM_HCC__ -D__HIP_PLATFORM_AMD__ -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_1_X_2024-06-22-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_1_X_2024-06-22-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_1_X_2024-06-22-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/alpaka/1.1.0-0a6641b4bfdf883c8da5b9e8620be504/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/boost/1.80.0-a1544032d9d65904ac2112b6d35bba55/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/lcg/root/6.30.07-5b5d83179d6ed7bc60aa7a5006899eba/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/tbb/v2021.9.0-1dc6d66b4b014f4ae4733b04914ce7d4/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/fmt/8.0.1-258b4791803c34b7e98cf43693e54d87/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/rocm/5.6.1-9b7217069df29cdf1c0bc74e165c09ec/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02842/el8_amd64_gcc12/external/tinyxml2/6.2.0-88fe0ec301baf763fa3c485e5b67ed91/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++17 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DALPAKA_DEFAULT_HOST_MEMORY_ALIGNMENT=128 -DALPAKA_DISABLE_VENDOR_RNG -DALPAKA_ACC_GPU_HIP_ENABLED -DALPAKA_ACC_GPU_HIP_ONLY_MODE -DALPAKA_HOST_ONLY -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestROCmAsync/ES_AlpakaESTestData.cc.d src/HeterogeneousCore/AlpakaTest/src/alpaka/ES_AlpakaESTestData.cc -o tmp/el8_amd64_gcc12/src/HeterogeneousCore/AlpakaTest/src/alpaka/HeterogeneousCoreAlpakaTestROCmAsync/ES_AlpakaESTestData.cc.o
</pre>

--- a/HeterogeneousCore/AlpakaTest/BuildFile.xml
+++ b/HeterogeneousCore/AlpakaTest/BuildFile.xml
@@ -1,6 +1,14 @@
 <use name="FWCore/Framework"/>
 <use name="HeterogeneousCore/AlpakaCore"/>
 <use name="HeterogeneousCore/AlpakaInterface"/>
+<!--
+Let the alpaka libraries depend on the main library.
+The possible syntaxes are:
+  <use name="HeterogeneousCore/AlpakaTest" for="alpaka"/>
+  <use name="1" for="alpaka"/>
+  <flags USE_ALPAKA="1"/>
+-->
+<use name="HeterogeneousCore/AlpakaTest" for="alpaka"/>
 <flags ALPAKA_BACKENDS="1"/>
 <export>
   <lib name="1"/>

--- a/HeterogeneousCore/AlpakaTest/interface/HostOnlyType.h
+++ b/HeterogeneousCore/AlpakaTest/interface/HostOnlyType.h
@@ -1,0 +1,21 @@
+#ifndef HeterogeneousCore_AlpakaTest_interface_HostOnlyType_h
+#define HeterogeneousCore_AlpakaTest_interface_HostOnlyType_h
+
+namespace alpakatest {
+
+  /* A simple class to demonstarte the dependency on host-only types from alpaka libraries */
+  class HostOnlyType {
+  public:
+    HostOnlyType() : value_{0} {}
+    HostOnlyType(int value) : value_{value} {}
+    void set(int value) { value_ = value; }
+    int get() { return value_; }
+    void print();
+
+  private:
+    int value_;
+  };
+
+}  // namespace alpakatest
+
+#endif  // HeterogeneousCore_AlpakaTest_interface_HostOnlyType_h

--- a/HeterogeneousCore/AlpakaTest/interface/alpaka/printAnswer.h
+++ b/HeterogeneousCore/AlpakaTest/interface/alpaka/printAnswer.h
@@ -1,0 +1,12 @@
+#ifndef HeterogeneousCore_AlpakaTest_interface_alpaka_printAnswer_h
+#define HeterogeneousCore_AlpakaTest_interface_alpaka_printAnswer_h
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::alpakatest {
+
+  void printAnswer();
+
+}  // namespace ALPAKA_ACCELERATOR_NAMESPACE::alpakatest
+
+#endif  // HeterogeneousCore_AlpakaTest_interface_alpaka_printAnswer_h

--- a/HeterogeneousCore/AlpakaTest/src/HostOnlyType.cc
+++ b/HeterogeneousCore/AlpakaTest/src/HostOnlyType.cc
@@ -1,0 +1,9 @@
+#include <iostream>
+
+#include "HeterogeneousCore/AlpakaTest/interface/HostOnlyType.h"
+
+namespace alpakatest {
+
+  void HostOnlyType::print() { std::cout << "The HostOnlyType value is " << value_ << '\n'; }
+
+}  // namespace alpakatest

--- a/HeterogeneousCore/AlpakaTest/src/alpaka/printAnswer.cc
+++ b/HeterogeneousCore/AlpakaTest/src/alpaka/printAnswer.cc
@@ -1,0 +1,14 @@
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaTest/interface/HostOnlyType.h"
+#include "HeterogeneousCore/AlpakaTest/interface/alpaka/printAnswer.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE::alpakatest {
+  using namespace ::alpakatest;
+
+  // A simple function to demonstarte the dependency on host-only types from alpaka libraries
+  void printAnswer() {
+    HostOnlyType answer(42);
+    answer.print();
+  }
+
+}  // namespace ALPAKA_ACCELERATOR_NAMESPACE::alpakatest

--- a/HeterogeneousCore/AlpakaTest/test/BuildFile.xml
+++ b/HeterogeneousCore/AlpakaTest/test/BuildFile.xml
@@ -13,3 +13,9 @@
   <!-- dependence only to trigger the unit test when AMD GPU is (expected to be) present -->
   <use name="rocm"/>
 </test>
+
+<bin name="alpakaTestPrintAnswer" file="alpaka/printAnswer.cpp">
+  <use name="HeterogeneousCore/AlpakaInterface"/>
+  <use name='HeterogeneousCore/AlpakaTest'/>
+  <flags ALPAKA_BACKENDS="1"/>
+</bin>

--- a/HeterogeneousCore/AlpakaTest/test/alpaka/printAnswer.cpp
+++ b/HeterogeneousCore/AlpakaTest/test/alpaka/printAnswer.cpp
@@ -1,0 +1,8 @@
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaTest/interface/alpaka/printAnswer.h"
+
+// each test binary is built for a single Alpaka backend
+using namespace ALPAKA_ACCELERATOR_NAMESPACE;
+using namespace ALPAKA_ACCELERATOR_NAMESPACE::alpakatest;
+
+int main() { printAnswer(); }

